### ISSUE TITLE
[custom cluster] adding 1 node all roles to custom rke2 cluster fails with `unable to create impersonator account: error getting service account token`

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#35593]
**Rancher Server Setup**
- Rancher version: v2.6-head (936c182)
- Installation option (Docker install/Helm Chart): docker
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:n/a

**Information about the Cluster**
- Kubernetes version:  v1.21.6+rke2r1

- Cluster Type (Local/Downstream):
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): custom (with aws node)

**Describe the bug**

when adding the first node to the custom custer (all roles), node begins to register but fails with an impersonation error. 

**To Reproduce**
 
* deploy a custom cluster on rancher
* run the command from the previous step (all roles should be selected) on a node/vm

**Result**
fails to add node with the following error in rancher:
```
Error while applying agent YAML, it will be retried automatically: exit status 1, Error from server (InternalError): an error on the server ("unable to create impersonator account: error getting service account token: error getting secret: secrets \"cattle-impersonation-system/cattle-impersonation-u-djxin436by-token-9vgqz\" not found") has prevented the request from succeeding
```

**Expected Result**

* custom cluster should allow registration of nodes and nodes should come to an active state

**Comment by cmurphy** [comment]
Is this reliably reproducible?

**Comment by cmurphy** [comment]
Hi @shmutalov, that looks like a different error message, the error message in this bug was `... error getting service account token: error getting secret:...` and the bug was introduced pre-release, while the error in your logs is `...error setting up impersonation for user u-vfc754tywx: Unauthorized```. Could you open up a separate bug with more information about your setup and what was happening when this error occurred?

**Comment by shmutalov** [comment]
@cmurphy I am getting that bug with rancher:latest and `rancher:v2.6-9740d7ab5b988456ad34423d81dba28c9640bb6c-head`. I am trying to create single node instance with known CA issued certificate.

```
docker run -d --privileged --restart=unless-stopped --name=rancher -p 1080:80 -p 1443:443 -v /opt/rancher:/var/lib/rancher -v /etc/ssl/certs/xxx-ml.crt:/etc/rancher/ssl/cert.pem -v /etc/ssl/private/xxx-ml.key:/etc/rancher/ssl/key.pem -v /etc/ssl/certs/xxx-ml.cabundle:/etc/rancher/ssl/cacerts.pem rancher/rancher
```

After server starts I am trying to create custom cluster:

```
sudo docker run -d --privileged --name=rancher-agent --restart=unless-stopped --net=host -v /etc/kubernetes:/etc/kubernetes -v /var/run:/var/run rancher/rancher-agent:VVV --server https://xxx.ml:1443 --token [KEKPUK] --ca-checksum [CHEBUREK] --etcd --controlplane --worker
```


My OS: 

```
shmutalov@laptop:~$ uname -a
Linux laptop 5.10.0-9-amd64 #1 SMP Debian 5.10.70-1 (2021-09-30) x86_64 GNU/Linux
shmutalov@laptop:~$ cat /etc/*rele*
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

**Comment by cmurphy** [comment]
@shmutalov I was not able to reproduce this following the steps you provided. Based on the log `failed to start cluster controllers c-58vmp: context canceled` and the `Unauthorized` problem I think something might have gone wrong on the agent. Are there any logs on the cluster agent that look like they could be related?

I found https://github.com/rancher/rancher/issues/30772 which looks similar, is it possible that your self-signed certs are causing problems? Does this work when you use the rancher-generated certs?

Since you're using the pre-released version, have you tried this in the past on earlier pre-released versions? Did this work for you on v2.6.2?

**Comment by slickwarren** [comment]
Tested on v2.6-head (de55fe0):
* deploy custom rke2 cluster as admin (single node - all roles)-- pass
  * add a node to the cluster -- pass
* deploy custom rke2 cluster as standard user -- pass
  * add a node to the cluster -- pass
